### PR TITLE
Fix aad configuration properties metadata and additional-spring-configuration-metadata

### PR DESCRIPTION
**Pull request by saragluna**
Some metadata are missing from the AAD and AAD B2C configuration properties, this PR will fix it. And some broken replacements in `additional-spring-configuration-metadata.json` file when changing properties in codes.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapp/AadAccessTokenGroupRolesExtractionTests.java [hunk 1]
```diff
@@ -61,11 +61,7 @@ void close() throws Exception {
     }
 
     private AadAuthenticationProperties getProperties() {
-        AadAuthenticationProperties properties = new AadAuthenticationProperties();
-        AadAuthenticationProperties.UserGroupProperties userGroup =
-            new AadAuthenticationProperties.UserGroupProperties();
-        properties.setUserGroup(userGroup);
-        return properties;
+        return new AadAuthenticationProperties();
```
Comment (moarychan) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapp/AadAccessTokenGroupRolesExtractionTests.java:64]: Due to no additional properties setting, this method can be removed.

**Review by moarychan — APPROVED** [review]
LGTM.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java [hunk 2]
```diff
@@ -43,19 +43,20 @@
      * Profile of Azure cloud environment.
      */
     @NestedConfigurationProperty
-    private AadProfileProperties profile = new AadProfileProperties();
+    private final AadProfileProperties profile = new AadProfileProperties();
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java:46]: Why `final` is necessary?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java [hunk 3]
```diff
@@ -193,7 +178,7 @@ public void setApplicationType(AadApplicationType applicationType) {
         /**
          * If "true", use "v1.0/me/transitiveMemberOf" to get members. Otherwise, use "v1.0/me/memberOf".
          */
-        private Boolean useTransitiveMembers = false;
+        private boolean useTransitiveMembers = false;
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java:181]: Why this change is necessary?

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/properties/AadAuthenticationProperties.java:181]: there's no need for using a Boolean, we already set it with a default value false, and the get method for it is   
```
public boolean isUseTransitiveMembers() {
            return useTransitiveMembers;
        }
```. If it's a Boolean, chances are it will throw NPE.

**Review by chenrujun — APPROVED** [review]
LGTM.
